### ISSUE TITLE
`one will omit a namespace`

### ISSUE THREAD

**Issue by yogat3ch** [yonicd/sinew#55]
Hi @yonicd, 
Thanks for the work on this useful package!
It looks like the text for the namespace selection indicates that selecting `1` will omit a namespace for the function in question, however, it does not actually do so - it simply applies the namespace in the `1` index from the selection options. 
The line where the text for the prompt is supplied is [here](https://github.com/yonicd/sinew/blob/32ada24ed695c60e79fbc44da21dbc42ed65b51a/R/pretty_utils.R#L171).

It seems like the expected functionality should probably ask for `0` , `NA`, or just blank to omit a namespace (to avoid confusion with option 1) and then actually omit a namespace for the function in question when `0`, `NA` or nothing is supplied. Currently, I tried all of these approaches and it appears to error and there's no way to omit a namespace at present.

**Comment by yonicd** [comment]
Thanks for the feedback. I'll take a look at this. 

Can you paste the menu you are seeing and the sessioninfo?

**Comment by yogat3ch** [comment]
Certainly,
Menu looks basically like this (with different function names):
```
Select which namespace to use for "validate"
(*) if you want it to persist for all subsequent instances
one will omit a namespace 
1: geojsonio::validate(*)
 2: jsonlite::validate(*)
 3: papaja::validate(*)
 4: R.utils::validate(*)
 5: rms::validate(*)
 6: shiny::validate(*)
 7: survival::validate(*)
 8: geojsonio::validate
 9: jsonlite::validate
10: papaja::validate
11: R.utils::validate
12: rms::validate
13: shiny::validate
14: survival::validate
```
SessionInfo:
```
R version 3.5.3 (2019-03-11)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19041)

Matrix products: default

locale:
[1] LC_COLLATE=English_United States.1252 
[2] LC_CTYPE=English_United States.1252   
[3] LC_MONETARY=English_United States.1252
[4] LC_NUMERIC=C                          
[5] LC_TIME=English_United States.1252    

attached base packages:
 [1] tools     grid      compiler  stats    
 [5] graphics  grDevices utils     datasets 
 [9] methods   base     

other attached packages:
 [1] gargle_0.5.0.9000      pkgload_1.1.0         
 [3] testthat_2.3.2         fs_1.4.1.9000         
 [5] desc_1.2.0             backports_1.1.6       
 [7] rprojroot_1.3-2        withr_2.3.0           
 [9] haven_2.2.0            colorspace_1.4-1      
[11] yaml_2.2.1             abind_1.4-7           
[13] hms_0.5.3              purrr_0.3.4.9000      
[15] glue_1.4.2             forcats_0.4.0         
[17] rematch2_2.1.2         RColorBrewer_1.1-2    
[19] openxlsx_4.1.4         vctrs_0.3.4.9000      
[21] generics_0.0.2         ellipsis_0.3.1        
[23] sos_2.0-2              brew_1.0-6            
[25] xml2_1.3.2             remotes_2.2.0         
[27] ggsignif_0.6.0         carData_3.0-3         
[29] stringi_1.5.4          cli_2.1.0             
[31] zip_2.0.4              scales_1.1.1.9000     
[33] magrittr_1.5           lifecycle_0.2.0.9000  
[35] gtable_0.3.0           ggpubr_0.2.4          
[37] dplyr_1.0.0            crayon_1.3.4          
[39] fansi_0.4.1            rio_0.5.16            
[41] gridExtra_2.3          tibble_3.0.4.9000     
[43] tidyselect_1.1.0       htmltools_0.5.0.9002  
[45] slickR_0.4.9           base64enc_0.1-3       
[47] pkgconfig_2.0.3        munsell_0.5.0         
[49] htmlwidgets_1.5.1.9001 foreign_0.8-71        
[51] stringr_1.4.0          readr_1.3.1           
[53] sinew_0.3.9            DT_0.15               
[55] car_3.0-5              data.table_1.13.0     
[57] rstudioapi_0.11        readxl_1.3.1          
[59] curl_4.3               rlang_0.4.8.9000      
[61] pillar_1.4.99.9002     ggplot2_3.3.2         
[63] httr_1.4.2             cellranger_1.1.0      
[65] R6_2.5.0               packrat_0.5.0         
[67] digest_0.6.26          assertthat_0.2.1      
[69] tidyr_1.1.0            HDA_0.0.0.9000        
[71] lubridate_1.7.9        Rcpp_1.0.5.4          
[73] RevoUtils_11.0.3       RevoUtilsMath_11.0.0 
```

**Comment by yonicd** [comment]
it's a typo... looks like the intention was to write none.

**Comment by yogat3ch** [comment]
That makes sense. If I just hit enter (none) I get:
`Enter an item from the menu, or 0 to exit`
It doesn't appear to be a way to move forward unless I hit 0 and break the operation altogether.  
When none is selected, `menu` returns 0 so I think there needs to be some logic (which is currently missing) to make it skip when `0` is the result

**Comment by yonicd** [comment]
thanks for catching this.

i think i didnt enter an offramp since there is an implied need to add an ns if you are running the function. 

I can add that option. you can PR if you want to, it should be pretty straight fw.

**Comment by yogat3ch** [comment]
The use case for the offramp is when the user has defined a custom function in their script that happens to match a function in a library. 

I can take a look at it, though I'm not quite sure how all the pieces of the package fit together, so it will take me a bit to understand what needs to be done in the 0 or none instance.

**Comment by yogat3ch** [comment]
Hi @yonicd,
I just submitted two PRs.
One for this issue entitled `choices_0_offramp`.
I also noticed that when `pretty_rmd_inline` is used to parse the chunks in an Rmd, it uses the tmpfile names when prompting the user with choices for functions which don't really allow the user to understand the context. However, when `create_library = TRUE` it uses the chunk names for some reason. 
I added functionality to pass chunk numbers and titles as names to the list with line number subsets. These are then passed on to the temp file names, and then shown to the user when prompted to choose a namespace.
